src/public/Interfaces.ts

```diff
@@ -1,21 +1,46 @@
+export interface IFunction {
+    (context: IContext, ...args: any[]): void;
+}
+
 export interface IContext {
     invocationId: string;
     executionContext: IExecutionContext;
     bindings: { [key: string]: any };
     bindingData: { [key: string]: any };
-    bindingDefinitions: { [key: string]: any };
+    bindingDefinitions: IBindingDefinition[];
     log: ILogger;
-    req?: IRequest;
-    res?: IResponse;
     done: IDoneCallback;
-};
+}
+
+export interface IHttpContext extends IContext {
+    req: IRequest;
```

Comment:
Should we just have IRequest? and leave res in IContext? Is it possible to define IResponse later after we figure out the right format?